File: api/Portfolio.Domain/Models/BlogPost.cs
using Portfolio.Domain.Models.Common;
using System;
using System.Collections.Generic;

namespace Portfolio.Domain.Models;

public class BlogPost : BaseEntity, IHasDisplayNumber, IFullyAudited
{
    #region Properties

    public string Title { get; set; }

    public string Description { get; set; }

    public string Content { get; set; }

    public bool IsPublished { get; set; }

    public int DisplayNumber { get; set; }

    public string MetaTitle { get; set; }

    public string MetaDescription { get; set; }

    public DateTime CreatedAtUTC { get; set; }

    public DateTime UpdatedAtUtc { get; set; }

    public Picture? Thumbnail { get; set; }

    public int? ThumbnailId { get; set; }

    public Picture? BannerPicture { get; set; }

    public int? BannerPictureId { get; set; }

    public ICollection<Comment> Comments { get; set; }

    #endregion
}

File: angular/projects/shared/src/lib/data/blog/blog-post.ts
import { Picture } from "../common/picture";
import { BaseBlog } from "./base-blog";
import { BlogComment } from "./comment";

export interface BlogPost extends BaseBlog {
    content: string | null;
    id: number;
    createdAtUtc: Date;
    updatedAtUtc: Date;
    metaTitle: string | null;
    metaDescription:  string | null;
    bannerPicture: Picture | null;
    thumbnail: Picture | null;
    comments: BlogComment[] | null;
}